apache2/msc_multipart.c

```diff
@@ -214,23 +214,42 @@ static int multipart_parse_content_disposition(modsec_rec *msr, char *c_d_value)
                 }
                 msr->multipart_filename = decoded_filename;
 
+               
+                if (msr->mpd->mpp->filenameext != NULL) {
+                    msr_log(msr, 4, "Multipart: Warning: Duplicate Content-Disposition filename*: %s",
+                     log_escape_nq(msr->mp, decoded_filename));
+                     return -17;
+                }
+
+                msr->mpd->mpp->filenameext = apr_pstrdup(msr->mp, decoded_filename);
+
+                // The `filename*` RCF 5987 encoded filename always overrides the `filename` parameter in content-disposition header.
+                msr->mpd->mpp->filename = msr->mpd->mpp->filenameext;
+
                 // Make sure to turn of INVALID quoting since RFC 5987 expects quotes in the filename format.
                 msr->mpd->flag_invalid_quoting = 0;
             }
             else
             {
-                decoded_filename = value;
-                validate_quotes(msr, value);
-                msr->multipart_filename = apr_pstrdup(msr->mp, decoded_filename);
-            }
+                // Process the `filename` attribute in the content-disposition header only if `filename*` does not exist.
+                if (msr->mpd->mpp->filenameext == NULL)
+                {
+                    // "name == 'filename'"
+                    decoded_filename = value;
+                    validate_quotes(msr, value);
```

Comment:
is this processing not needed on filenameext?

Reply:
Yeah, that's a good catch. I think we should run this validation on `filenameext`.

Reply:
Added the quote valdiation for the extended attributes.

Reply:
Fixed.